storage/pipeline/input.go

```diff
@@ -146,13 +151,24 @@ func (m *Sealing) maybeStartSealing(ctx statemachine.Context, sector SectorInfo,
 
 		var dealSafeSealEpoch abi.ChainEpoch
 		for _, piece := range sector.Pieces {
-			if piece.DealInfo == nil {
+			if !piece.HasDealInfo() {
 				continue
 			}
 
-			dealSafeSealEpoch = piece.DealInfo.DealProposal.StartEpoch - cfg.StartEpochSealingBuffer
+			startEpoch, err := piece.StartEpoch()
```

Comment:
later should check if there is deal term attached, if not just do sector stuff as it goes